Sources/Flare/Classes/Models/Internal/Protocols/IRenewalInfo.swift

```diff
@@ -44,3 +50,14 @@ public protocol IRenewalInfo {
     /// offers and if there will be no offer applied for the next billing period.
     var offerID: String? { get }
 }
+
+/// Default implementation of the currency property for backward compatibility.
+extension IRenewalInfo {
+    var currency: String? {
+        if #available(iOS 16.0, macOS 13.0, watchOS 9.0, tvOS 16.0, *) {
+            return Locale.current.currency?.identifier
+        } else {
+            return Locale.current.currencyCode
+        }
+    }
+}
```

Comment:
Could you create an extension for `Locale`? It seems that there's code duplication with your previous PR.